plugins/modules/oneview_firmware_bundle.py

```diff
@@ -121,10 +122,28 @@ def __present(self, file_path):
             return dict(changed=False, msg=self.MSG_ALREADY_PRESENT, ansible_facts=dict(firmware_bundle=self.current_resource.data))
 
     def __add_compsig(self, file_path):
-        if self.current_resource and self.current_resource.data.get('signatureFileRequired') is False:
-            self.current_resource = self.resource_client.upload_compsig(file_path)
+        file_name = basename(file_path)
+
+        if self.current_resource and self.current_resource.data.get('resourceState') == 'AddFailed' and\
+                self.current_resource.data.get('signatureFileRequired') is False:
+
+            self.current_resource = self.resource_client.upload_compsig(
+                file_path)
+            return dict(changed=True, msg=self.MSG_ADD_SIG, ansible_facts=dict(compsig=self.current_resource))
+
+        elif self.current_resource and self.current_resource.data.get('resourceState') == 'AddFailed' and\
+                file_name not in self.current_resource.data.get('signatureFileName'):
+
+            self.current_resource = self.resource_client.upload_compsig(
+                file_path)
             return dict(changed=True, msg=self.MSG_ADD_SIG, ansible_facts=dict(compsig=self.current_resource))
-        elif self.current_resource and self.current_resource.data.get('signatureFileRequired') is True:
+
+        elif self.current_resource and self.current_resource.data.get('resourceState') == 'AddFailed' and\
```

Comment:
do we need to different checks to see if file is already present. I mean 
if self.current_resource and file_name in self.current_resource.data.get('signatureFileName'):
will not do?